src/util/string.cpp

```diff
@@ -144,6 +144,16 @@ std::string wide_to_utf8(const std::wstring &input)
 	return out;
 }
 
+void wide_add_codepoint(std::wstring &result, char32_t codepoint)
+{
+	if ((0xD800 <= codepoint && codepoint <= 0xDFFF) || (0x10FFFF < codepoint)) {
```

Comment:
Couldn't the code for checking if the code point is valid be shared between the two implementations?

Reply:
Probably, I can change that if you want. However, since the code on WIN32 also compares the codepoint to 0x10000 in all cases, I thought it was more efficient to separate the condition.

Reply:
I imagine the compiler would happily do that on it's own, but I suppose this is perfectly fine too. It's not like valid unicode codepoints will change any time soon.